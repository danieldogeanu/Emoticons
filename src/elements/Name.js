import React from 'react';

function Name(props) {
	return (
		<span className="Name">
			{props.name}
		</span>
	);
}

export default Name;
