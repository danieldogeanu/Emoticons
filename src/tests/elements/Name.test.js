import React from 'react';
import {render} from '@testing-library/react';
import Name from '../../elements/Name';

const testName = 'Test Emoticon Name';

describe('Name Element', () => {

	it('renders emoticon name properly', () => {
		const {getByText} = render(<Name name={testName} />);
		const renderedName = getByText(testName);

		expect(renderedName).toBeInTheDocument();
		expect(renderedName).toHaveTextContent(testName);
		expect(renderedName).toHaveClass('Name');
		expect(renderedName).toMatchSnapshot();
	});

});
