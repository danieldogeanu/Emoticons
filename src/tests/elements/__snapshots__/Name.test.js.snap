// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name Element renders emoticon name properly 1`] = `
<span
  class="Name"
>
  Test Emoticon Name
</span>
`;
